zad4/app/models/player.rb

```diff
@@ -0,0 +1,4 @@
+class Player < ApplicationRecord
+  validates :first_name, :last_name
```

Comment:
Tu zapomniałeś podać typ walidacji presence ;) Skutkuje to błędem:
```
An error occurred while loading ./spec/models/player_spec.rb.
Failure/Error: validates :first_name, :last_name

ArgumentError:
  You need to supply at least one validation
```

Wszystkie walidacje można to zapisać rozdzielnie:
```ruby
class Player < ActiveRecord::Base
  validates :first_name, presence: true
  validates :last_name, presence: true
end
```

Lub łącznie:
```ruby
class Player < ActiveRecord::Base
  validates :first_name,  :last_name, presence: true
end
```

Obie wersję działają, ale na codzień w praktyce jest walidowane trochę więcej rzeczy niż presence, atrybutów jest więcej niż 3 i rozpisywanie walidacji oddzielnie wygląda czytelniej